ethcontract/src/contract/method.rs

```diff
@@ -316,14 +270,15 @@ async fn convert_response<
 /// encode this information in a JSON RPC error. On a revert or invalid opcode,
 /// the result is `0x` (empty data), while on a revert with message, it is an
 /// ABI encoded `Error(string)` function call data.
-fn decode_geth_call_result<R: Detokenizable>(
+fn decode_geth_call_result<R: Tokenize>(
     function: &Function,
     bytes: Vec<u8>,
-) -> Result<R::Output, ExecutionError> {
+) -> Result<R, ExecutionError> {
+    let is_void = R::from_token(Token::Tuple(Vec::new())).is_ok();
```

Comment:
Maybe this can be part of the `Tokenize` trait as a default method?

Although, on second thought, this might no longer be needed with #477, so maybe just leave it as it currently is.